components/sidebar.jsx

```diff
@@ -5,27 +5,85 @@ var LinkAnchorSwap = require('./link-anchor-swap.jsx');
 
 var Login = require('./login.jsx');
 
+var Subitems = React.createClass({
+  render: function() {
+    var items = this.props.subItems.map(function (item, key) {
+      return (
+        <li key={item.name}>
+          <Link to={item.link}>
+            {item.name}
+          </Link>
+        </li>
+      )}
+    );
+    return (
+      <ul className="sidebar-subitems">
+        {items}
+      </ul>
+    );
+  }
+});
+
+var TopLevelNavItem = React.createClass({
+  getInitialState: function() {
+    return {
+      activeSubNav: false
+    };
+  },
+  componentDidMount: function() {
+    this.highlightActiveSubNav();
+  },
+  componentDidUpdate: function() {
```

Comment:
ha, totally forgot that they come with oldProps, oldState arguments! Thanks for reminding me!